src/App.tsx

```diff
@@ -1,14 +1,37 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import './App.scss';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { FilterType } from './types/FilterTypes';
+import { preparedTodos } from './utils/FilterFunction';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [filterType, setFiltertype] = useState(FilterType.All);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const visibleTodos = preparedTodos(todos, query, filterType);
```

Comment:
Use useMemo here to avoid filtering todos on every render ( e.g if selectedTodo changed)